### ISSUE TITLE
IOException: Unknown GeoDistance ordinal [2]

### ISSUE THREAD

**Issue by mloho** [elastic/elasticsearch#29301]
**Elasticsearch server version**: `{"number": "5.6.6","lucene_version": "6.6.1"}`
**Elasticsearch client version**: `6.2.1`

**Plugins installed**: `[{
"name": "x-pack",
"version": "5.6.6",
"description": "Elasticsearch Expanded Pack Plugin",
"classname": "org.elasticsearch.xpack.XPackPlugin",
"has_native_controller": true
}]`

**JVM version**: `1.8.0_151`

**OS version**: Linux ubuntu 4.4.0-116-generic

**Description of the problem including expected versus actual behavior**:
When I include a geo distance query into the should clause of my master bool query and use this query to search for results, it would produce a `java.io.IOException: Unknown GeoDistance ordinal [2]`
*Expected*: search to return with correct results. *Actual*: search produces an IOException.

**Notes & Observations**:
- The master bool query also contains other queries in its other clauses. However, the search with the same parameters works when the geo distance query is excluded.
- The search is successful if I manually copy the master bool query string and paste it into my ElasticSearch Head browser plugin (and also curl I'm assuming).

**Steps to reproduce**:

```java
BoolQueryBuilder queryBuilder = QueryBuilders.boolQuery();
queryBuilder.should(
    QueryBuilders.geoDistanceQuery(term).point(origin).distance(distance)
);

client.prepareSearch(index)
        .setTypes(type)
        .setFrom(from)
        .setSize(size)
        .setQuery(queryBuilder);
```

**Stack trace**:
[java.io.IOException Unknown GeoDistance ordinal [2].txt](https://github.com/elastic/elasticsearch/files/1861428/java.io.IOException.Unknown.GeoDistance.ordinal.2.txt)

**Comment by imotov** [comment]
It looks like we accidently broke backward compatibility [here](https://github.com/elastic/elasticsearch/commit/b1a6b227e18cd71de78b77810a284b3a8cc39175#diff-07ff0f9821841e51d05e48d81d599582L113). Unfortunately, I cannot think of any temporary workaround besides using a client and a server with the same major version.

**Comment by imotov** [comment]
@mloho, thanks for the detailed report. It was fixed by #29325 and should be release in v6.3.0.
